### PR TITLE
py-tldextract: new package

### PR DESCRIPTION
**Pull request by LydDeb**

### CODE REVIEW

**Comment by LydDeb** [comment]
Depends on [py-requests-file](https://github.com/spack/spack/pull/41331)

**Review by tldahlgren — COMMENTED** [review]
FYI. Confirmed homepage and version sha256.

**Review by manuelakuhn — DISMISSED** [review]
There is also a much newer version (5.1.1) available on Pypi in case you want to add this as well.

File: var/spack/repos/builtin/packages/py-tldextract/package.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    pypi = "tldextract/tldextract-3.4.1.tar.gz"
+
+    maintainers("LydDeb")
+
+    version("5.1.1", sha256="9b6dbf803cb5636397f0203d48541c0da8ba53babaf0e8a6feda2d88746813d4")
+    version("3.4.1", sha256="fa9e50c4a03bede2a1d95dca620d661678484626858ccf388cf9671a0dd497a4")
+
+    depends_on("py-setuptools@61.2:", when="@5.1.1", type="build")
+    depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools-scm@6.2+toml", when="@5.1.1", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tldextract/package.py]:
```suggestion
    depends_on("py-setuptools-scm@6.2:+toml", when="@5.1.1", type="build")
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**